src/thunderbots/software/test/world/field.cpp

```diff
@@ -14,20 +14,139 @@ TEST(FieldTest, construction)
+
+    EXPECT_EQ(Point(-4.5, 0.5), field.friendlyGoalpostPos());
+    EXPECT_EQ(Point(-4.5, -0.5), field.friendlyGoalpostNeg());
+    EXPECT_EQ(Point(4.5, 0.5), field.enemyGoalpostPos());
+    EXPECT_EQ(Point(4.5, -0.5), field.enemyGoalpostNeg());
+
+    EXPECT_EQ(Rect(Point(-4.5, 1.0), Point(-3.5, -1.0)), field.friendlyDefenseArea());
+    EXPECT_EQ(Rect(Point(4.5, 1.0), Point(3.5, -1.0)), field.enemyDefenseArea());
+
+    EXPECT_EQ(Point(-3.5, 0.0), field.penaltyFriendly());
+    EXPECT_EQ(Point(3.5, 0.0), field.penaltyEnemy());
+
+    EXPECT_EQ(Point(-4.5, 3.0), field.friendlyCornerPos());
+    EXPECT_EQ(Point(-4.5, -3.0), field.friendlyCornerNeg());
+    EXPECT_EQ(Point(4.5, 3.0), field.enemyCornerPos());
+    EXPECT_EQ(Point(4.5, -3.0), field.enemyCornerNeg());
+}
+
+TEST(FieldTest, update_specific_params)
+{
+    Field field                 = Field();
+    double length               = 9.0;
+    double width                = 6.0;
     double defense_width        = 2.0;
     double defense_length       = 1.0;
+    double goal_width           = 1.0;
     double boundary_width       = 0.3;
     double center_circle_radius = 0.5;
 
     field.updateDimensions(length, width, defense_length, defense_width, goal_width,
                            boundary_width, center_circle_radius);
 
+    field.updateDimensions(field.length(), width / 2, field.defenseAreaLength(),
+                           defense_width * 1.5, field.goalWidth(), boundary_width * 0,
+                           field.centreCircleRadius());
+
+    EXPECT_TRUE(field.valid());
+
+    EXPECT_DOUBLE_EQ(9.0, field.totalLength());
+    EXPECT_DOUBLE_EQ(3.0, field.totalWidth());
+    EXPECT_DOUBLE_EQ(0.0, field.boundaryWidth());
+
+    EXPECT_EQ(Point(-4.5, 0.0), field.friendlyGoal());
+    EXPECT_EQ(Point(4.5, 0.0), field.enemyGoal());
+
+    EXPECT_EQ(Point(-4.5, 0.5), field.friendlyGoalpostPos());
+    EXPECT_EQ(Point(-4.5, -0.5), field.friendlyGoalpostNeg());
+    EXPECT_EQ(Point(4.5, 0.5), field.enemyGoalpostPos());
+    EXPECT_EQ(Point(4.5, -0.5), field.enemyGoalpostNeg());
+
+    EXPECT_EQ(Rect(Point(-4.5, 1.5), Point(-3.5, -1.5)), field.friendlyDefenseArea());
+    EXPECT_EQ(Rect(Point(4.5, 1.5), Point(3.5, -1.5)), field.enemyDefenseArea());
+
+    EXPECT_EQ(Point(-3.5, 0.0), field.penaltyFriendly());
+    EXPECT_EQ(Point(3.5, 0.0), field.penaltyEnemy());
+
+    EXPECT_EQ(Point(-4.5, 1.5), field.friendlyCornerPos());
+    EXPECT_EQ(Point(-4.5, -1.5), field.friendlyCornerNeg());
+    EXPECT_EQ(Point(4.5, 1.5), field.enemyCornerPos());
+    EXPECT_EQ(Point(4.5, -1.5), field.enemyCornerNeg());
+
+    // Repeat with a different combination of parameters
+    field.updateDimensions(field.length() / 3, field.width(),
+                           field.defenseAreaLength() * 0.5, field.defenseAreaWidth(),
+                           field.goalWidth() * 2, field.boundaryWidth(),
+                           field.centreCircleRadius() * 1.25);
```

Comment:
Should maybe be a different test?

Reply:
Part of the purpose of the test was to make sure there were no side effects, which is why I left the two different "steps" together. It should be fine to split thought because it doesn't matter what the values are beforehand. Will fix tonight.

Reply:
done